app/controllers/my_modules_controller.rb

```diff
@@ -455,25 +462,24 @@ def assign_repository_records
 
   def unassign_repository_records
     if params[:selected_rows].present? && params[:repository_id].present?
-      records = []
       downstream = ActiveModel::Type::Boolean.new.cast(params[:downstream])
 
-      params[:selected_rows].each do |id|
-        record = RepositoryRow.find_by_id(id)
-        next unless record && @my_module.repository_rows.include?(record)
-        record.last_modified_by = current_user
-        record.save
-        records << record
-      end
+      records = RepositoryRow.assigned_on_my_module(params[:selected_rows],
+                                                    @my_module)
 
       @my_module.repository_rows.destroy(records & @my_module.repository_rows)
 
       if downstream
         @my_module.downstream_modules.each do |my_module|
+          records = RepositoryRow.assigned_on_my_module(params[:selected_rows],
+                                                        my_module)
           my_module.repository_rows.destroy(records & my_module.repository_rows)
         end
       end
 
+      # update last last_modified_by
+      records.update_all(last_modified_by_id: current_user.id)
```

Comment:
Hmm, this `records` variable will probably hold the latest `records` collection from the last downstream module. So I doubt it's the right array of records to be used for setting `last_modified_by_id`.

Probably, you need to somehow have a collection/hash/map of all destroyed repository rows (both in initial module/task, as well as in downstream modules/tasks), and update all such records.